keras/applications/resnet_rs.py

```diff
@@ -0,0 +1,891 @@
+                mean=[0.485, 0.456, 0.406],
+                variance=[0.229**2, 0.224**2, 0.225**2],
+                axis=bn_axis,
+            )(x)
+        else:
+            x = layers.Rescaling(scale=1. / 128.0, offset=-1)(x)
+
+    # Build stem
+    x = STEM(bn_momentum=bn_momentum, bn_epsilon=bn_epsilon, activation=activation)(
+        x
+    )
+
+    # Build blocks
+    if block_args is None:
+        block_args = BLOCK_ARGS[depth]
+
+    for i, args in enumerate(block_args):
+        survival_probability = get_survival_probability(
+            init_rate=drop_connect_rate,
+            block_num=i + 2,
+            total_blocks=len(block_args) + 1,
+        )
+
+        x = BlockGroup(
+            filters=args["input_filters"],
+            activation=activation,
+            strides=(1 if i == 0 else 2),
+            num_repeats=args["num_repeats"],
+            se_ratio=se_ratio,
+            bn_momentum=bn_momentum,
+            bn_epsilon=bn_epsilon,
+            survival_probability=survival_probability,
+            name=f"c{i + 2}_",
+        )(x)
+
+    # Build head:
+    if include_top:
+        x = layers.GlobalAveragePooling2D(name="avg_pool")(x)
+        if dropout_rate > 0:
+            x = layers.Dropout(dropout_rate, name="top_dropout")(x)
+
+        imagenet_utils.validate_activation(classifier_activation, weights)
+        x = layers.Dense(
+            classes, activation=classifier_activation, name="predictions"
+        )(x)
+    else:
+        if pooling == "avg":
+            x = layers.GlobalAveragePooling2D(name="avg_pool")(x)
+        elif pooling == "max":
+            x = layers.GlobalMaxPooling2D(name="max_pool")(x)
+
+    # Ensure that the model takes into account
+    # any potential predecessors of `input_tensor`.
+    if input_tensor is not None:
+        inputs = layer_utils.get_source_inputs(input_tensor)
+    else:
+        inputs = img_input
+
+    # Create model.
+    model = training.Model(inputs, x, name=model_name)
+
+    # Download weights
+    if weights in weights_allow_list:
+        weights_input_shape = weights.split("-")[-1]  # e. g. "i160"
+        weights_name = f"{model_name}-{weights_input_shape}"
+        if not include_top:
+            weights_name += "_notop"
+
+        filename = f"{weights_name}.h5"
+        download_url = BASE_WEIGHTS_URL + filename
+        weights_path = data_utils.get_file(
+            fname=filename,
+            origin=download_url,
+            cache_subdir="models",
+            file_hash=WEIGHT_HASHES[filename],
+        )
+        model.load_weights(weights_path)
+
+    elif weights is not None:
+        model.load_weights(weights)
+
+    return model
+
+@keras_export('keras.applications.resnet_rs.ResNetRS50',
+              'keras.applications.ResNetRS50')
+def ResNetRS50(
+    include_top=True,
+    weights="imagenet",
+    classes=1000,
+    input_shape=None,
+    input_tensor=None,
+    pooling=None,
+    classifier_activation="softmax",
```

Comment:
We should also include `include_preprocessing=True` in the signature for all publicly exported applications

Reply:
Done